packages/@aws-cdk/aws-ecr-assets/test/test.image-asset.ts

```diff
@@ -188,30 +188,51 @@ export = {
     const app = new App();
     const stack = new Stack(app, 'stack');
 
-    new DockerImageAsset(stack, 'MyAsset', {
+    const image = new DockerImageAsset(stack, 'MyAsset', {
       directory: path.join(__dirname, 'demo-image')
     });
 
     const session = app.synth();
 
-    test.ok(fs.existsSync(path.join(session.directory, 'asset.1a17a141505ac69144931fe263d130f4612251caa4bbbdaf68a44ed0f405439c/Dockerfile')));
-    test.ok(fs.existsSync(path.join(session.directory, 'asset.1a17a141505ac69144931fe263d130f4612251caa4bbbdaf68a44ed0f405439c/index.py')));
+    test.ok(fs.existsSync(path.join(session.directory, `asset.${image.sourceHash}`, 'Dockerfile')));
+    test.ok(fs.existsSync(path.join(session.directory, `asset.${image.sourceHash}`, 'index.py')));
     test.done();
   },
 
   'docker directory is staged without files specified in .dockerignore'(test: Test) {
     const app = new App();
     const stack = new Stack(app, 'stack');
 
-    new DockerImageAsset(stack, 'MyAsset', {
+    const image = new DockerImageAsset(stack, 'MyAsset', {
       directory: path.join(__dirname, 'dockerignore-image')
     });
 
     const session = app.synth();
 
-    test.ok(fs.existsSync(path.join(session.directory, `asset.1a17a141505ac69144931fe263d130f4612251caa4bbbdaf68a44ed0f405439c/Dockerfile`)));
-    test.ok(fs.existsSync(path.join(session.directory, 'asset.1a17a141505ac69144931fe263d130f4612251caa4bbbdaf68a44ed0f405439c/index.py')));
-    test.ok(!fs.existsSync(path.join(session.directory, 'asset.1a17a141505ac69144931fe263d130f4612251caa4bbbdaf68a44ed0f405439c/foobar.txt')));
+    // .dockerignore itself should be included in output to be processed during docker build
+    test.ok(fs.existsSync(path.join(session.directory, `asset.${image.sourceHash}`, '.dockerignore')));
+    test.ok(fs.existsSync(path.join(session.directory, `asset.${image.sourceHash}`, `Dockerfile`)));
+    test.ok(fs.existsSync(path.join(session.directory, `asset.${image.sourceHash}`, 'index.py')));
+    test.ok(!fs.existsSync(path.join(session.directory, `asset.${image.sourceHash}`, 'foobar.txt')));
+
+    test.done();
+  },
+
+  'docker directory is staged without files specified in exclude option'(test: Test) {
+    const app = new App();
+    const stack = new Stack(app, 'stack');
+
+    const image = new DockerImageAsset(stack, 'MyAsset', {
+      directory: path.join(__dirname, 'dockerignore-image'),
+      exclude: ['.dockerignore']
```

Comment:
Just so it’s less confusing, can you exclude a different file in the test? Otherwise it might look to future us that “exclude” refers to an “ignore file” and not to a list of exclusion globs.

Reply:
Thank you for your review. I've modified the test case.